data/posts/rsc-framework-mode-preview.md

```diff
@@ -0,0 +1,117 @@
+
+Now that RSC Data Mode is in place, we're able to provide an RSC-powered Vite plugin that compiles to these lower level APIs.
+
+As noted in our post on ["React Router and React Server Components: The Path Forward,"](./react-router-and-react-server-components) the great thing about RSC is that this new Framework Mode plugin is much simpler than our earlier non-RSC work. Most of the framework-level complexity is now implemented at a lower level in RSC Data Mode, with RSC Framework Mode being a more lightweight layer on top.
+
+## Getting started
+
+Today, when using the regular non-RSC Framework Mode, you add the React Router plugin to your Vite config:
+
+<!-- prettier-ignore -->
+```ts
+// vite.config.ts
+import { defineConfig } from "vite";
+import { reactRouter } from "@react-router/dev/vite";
+
+export default defineConfig({
+  plugins: [
+    reactRouter(),
+  ],
+});
+```
+
+To enable RSC Framework Mode, you simply swap this out for the new `unstable_reactRouterRSC` Vite plugin, along with the official (experimental) [@vitejs/plugin-rsc](https://www.npmjs.com/package/@vitejs/plugin-rsc) as a peer dependency.
+
+```sh
+npm install @vitejs/plugin-rsc
+```
+
+<!-- prettier-ignore -->
+```ts
+// vite.config.ts
+import { defineConfig } from "vite";
+import { unstable_reactRouterRSC } from "@react-router/dev/vite";
+import { rsc } from "@vitejs/plugin-react";
+
+export default defineConfig({
+  plugins: [
+    unstable_reactRouterRSC(),
+    rsc(),
+  ],
+});
+```
+
+With these new plugins, all of the features described in our [RSC Data Mode preview blog post](./rsc-preview) will be available in your Framework Mode app:
+
+- The ability to return RSC content from loaders/actions.
+- RSC-first "Server Component Routes" by exporting a `ServerComponent` instead of the usual `default` component export.
+- Client components with the `"use client"` directive.
+- Server Functions with the `"use server"` directive.
+
+Note that some Framework Mode features are not available in our initial preview release. Most notably, SPA Mode, pre-rendering and custom build entries are not yet supported, but rest assured we plan on supporting these before a stable release.
+
+## Build output
+
+We've worked to keep the build output as close as possible to non-RSC Framework Mode. In fact, the top-level structure is exactly the same:
+
+- `build/server` - Server code
+- `build/client` - Browser assets
+
+What's changed is the content of `build/server/index.js`.
+
+In non-RSC Framework Mode, the exports from `build/server/index.js` form a React Router `ServerBuild` object. In order to make use of this, you need to pass it through an adapter to your desired runtime, e.g. `createRequestListener` from `@react-router/node`, or `createRequestHandler` from `@react-router/express`.
+
+In RSC Framework Mode, `build/server/index.js` exports a plain old request handler function (`(request: Request) => Promise<Response>`) designed for handling document/data requests.
```

Comment:
I think small code snippets would help clarify the change here

Non-RSC:

```ts
import { createRequestHandler } from "@react-router/express";
import express from "express";

const app = express();
const build = await import("./build/server/index.js");
app.all("*", createRequestHandler({ build }));
```

RSC:

```ts
import { createRequestListener } from "@remix-run/node-fetch-server";
import express from "express";

const app = express();
const build = await import("./build/server/index.js");
app.all("*", createRequestListener(build.default));
```